shared/login/reset/password.tsx

```diff
@@ -2,36 +2,108 @@ import * as React from 'react'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
 import {SignupScreen} from '../../signup/common'
+import HiddenString from '../../util/hidden-string'
+import * as Container from '../../util/container'
+import * as AutoresetGen from '../../actions/autoreset-gen'
+import * as Constants from '../../constants/autoreset'
+import {useWaiting} from '../../constants/waiting'
 
-const todo = () => console.log('todo')
-
-const KnowPassword = () => (
-  <SignupScreen
-    title="Account reset"
-    noBackground={true}
-    onBack={todo}
-    leftActionText="Cancel"
-    buttons={[{label: 'Yes', onClick: todo, type: 'Success'}, {label: 'No', onClick: todo, type: 'Dim'}]}
-  >
-    <Kb.Box2 direction="vertical" gap="medium" fullWidth={true} fullHeight={true} centerChildren={true}>
-      <Kb.Icon type="iconfont-password" color={Styles.globalColors.black} fontSize={24} />
-      <Kb.Box2 direction="vertical" centerChildren={true}>
-        <Kb.Text type="Header" center={true}>
-          Do you know your
-        </Kb.Text>
-        <Kb.Text type="Header">password?</Kb.Text>
+const KnowPassword = () => {
+  const error = Container.useSelector(state => state.autoreset.error)
+  const waiting = useWaiting(Constants.autoresetEnterPipelineWaitingKey)
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const onCancel = React.useCallback(() => dispatch(nav.safeNavigateUpPayload()), [])
+  const onYes = React.useCallback(
+    () => dispatch(nav.safeNavigateAppendPayload({path: ['resetEnterPassword']})),
+    []
+  )
+  const onNo = React.useCallback(() => dispatch(AutoresetGen.createResetAccount({})), [])
```

Comment:
are these `useCallback`s actually necessary?

Reply:
They're not **necessary**, but they allow some children to avoid updating when this component does. "Some children" includes connected components, things that use `React.memo`, I think `PureComponents`, some other stuff. We do pass the raw fcns in many places, but I think in new functional components we ought to `useCallback`.